torch/_dynamo/variables/torch.py

```diff
@@ -834,11 +834,24 @@ def handle_addcdiv(self, tx: "InstructionTranslator", *args, **kwargs):
         @register(torch.full)
         def handle_full(self, tx, size, fill_value, **kwargs):
             if isinstance(fill_value, TensorVariable):
-                result = TorchInGraphFunctionVariable(
-                    torch.ops.aten._local_scalar_dense
-                ).call_function(tx, [fill_value], {})
-                return TorchInGraphFunctionVariable(torch.full).call_function(
-                    tx, [size, result], kwargs
+                # Decompose: create empty tensor and fill it
+                # This avoids the scalar extraction at compile time
+                empty_result = TorchInGraphFunctionVariable(torch.empty).call_function(
+                    tx, [size], kwargs
+                )
+                # Call fill_ method on the empty tensor
+                return empty_result.call_method(tx, "fill_", [fill_value], {})
+            else:
+                # For Python scalars and other non-tensor types, use default lowering
```

Comment:
Ah, I thought this was supposed to default to the original logic:
```
                result = TorchInGraphFunctionVariable(
                    torch.ops.aten._local_scalar_dense
                ).call_function(tx, [fill_value], {})
                return TorchInGraphFunctionVariable(torch.full).call_function(
                    tx, [size, result], kwargs
```

Why did we change this to wrap_fx_proxy?

That is the source of the ci failure (arg `requires_grad` gets forwarded to the `aten.full.default` which doesn't exist